### PR TITLE
fix(rails): fix nil dereference in SQL Active Record `bind` key

### LINKED ISSUE #555: NoMethodError after updating to 6.11.1
### Description
I’m trying to update the bugsnag gem from 6.10.0 to 6.11.1 for a rails application and the following error started to pop up when running the test suite. I'm able to run 6.10.0 without any issues.

```
undefined method `each_with_object' for nil:NilClass (NoMethodError)
/gems/bugsnag-6.11.1/lib/bugsnag/integrations/railtie.rb:82:in `block in event_subscription'
```

Might be related to using `mysql2` as [discussed here](https://github.com/bugsnag/bugsnag-ruby/pull/521#discussion_r245392743).

Configuration for bugsnag is pretty minimal.

```
Bugsnag.configure do |config|
  config.api_key =
  config.meta_data_filters +=
  config.release_stage ||= ENV['RAILS_ENV'] || 'development'
  config.ignore_classes += [SignalException, SystemExit]
end
```

### Environment
* Rails 3.2.22
* Bugsnag 6.11.1
* mysql2 0.3.21

### PR DESCRIPTION
**Pull request by tomlongridge**
## Goal

Older versions of `mysql2` adapter do not contain the `bind` key in the Active Record notification.

### Changed

* `railtie.rb` - added extra guard for key not being present in notification

## Tests

I was unable to get a Rails instance with `mysql2` v0.3.x working to reproduce this issue. However I have run it with later versions to ensure that we pick up the `bind` key correctly still (this is also checked on MazeRunner tests).

### Linked issues

Fixes #555

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Changelog
 =========
 
+## TBC
+
+### Fixes
+
+* Fix nil dereference in SQL Active Record `bind` key.
```
Comment (tobyhs) [inline CHANGELOG.md]: `:binds` instead of `bind`, although I would probably reword this to something like "Account for missing `:binds` key in `sql.active_record` ActiveSupport notifications"
